massa-execution-worker/src/execution.rs

```diff
@@ -248,7 +242,75 @@ impl ExecutionState {
         if slot > max_slot {
             panic!("cannot execute at a slot beyond active cursor + 1");
         }
+    }
+
+    /// Lazily query (from end to beginning) the active balance of an address at a given slot.
+    /// Returns None if the address balance could not be determined from the active history.
+    pub fn get_active_balance_at_slot(&self, slot: Slot, addr: &Address) -> Option<Amount> {
```

Comment:
But this function is only used in [get_final_and_active_parallel_balance](https://github.com/massalabs/massa/blob/ledger-unification/massa-execution-worker/src/execution.rs#L835). Searching in the final ledger would mean doing twice the same operation. I think we should either search in the final ledger if `get_active_balance_at_slot` does not succeed and not return the final balance, or the other way around but not both.

Reply:
Oh ok then I think that this function should be renamed to reflect that we are only looking into the active history (we are not getting the absolute active value).

In turn, when looking for the "active balance", get_final_and_active_parallel_balance should return the info from the active history if available, otherwise it should return the final value

Reply:
To clarify: "active balance" does not mean "modified within the history". It means "the absolute balance at the output of the current slot". That means that if the active history does not give any info on it, it needs to be returned from the final state.